### PR TITLE
Refactor environment and upgrade to Gunicorn

### LINKED ISSUE #120: Refactor Local Environment Setup
A little spring cleaning, update the local environment setup tools to current recommended practices, including:

- Flask config files > Flask CLI
- Virtualenv > Pipenv

At the same time, creating better documentation for the implementation and use of these tools on setup and deployment should provide a good new base.

### PR DESCRIPTION
**Pull request by jaredbriskman**
This should resolve #120

### CODE REVIEW

**Comment by jaredbriskman** [comment]
After discussion, changed config to resolve #129 .

**Comment by newsch** [comment]
@jaredbriskman Looks like the review app has crashed, and had 4 boot timeouts. Looking through the logs, I'm seeing errors binding to a port and deprecation warnings from Celery.

In general, this looks good - well documented, pipenv _just works_, and gunicorn "seems faster". Let's meet up IRL to go over it and how to ensure a smooth transition for development - I'll read up on gunicorn in the meantime.

File: Pipfile [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+pymongo = ">=3.4.0"
+python-dateutil = ">=2.6.0"
+python-engineio = ">=1.7.0"
+python-socketio = ">=1.7.6"
+pytz = ">=2017.2"
+requests = ">=2.18.1"
+"urllib3" = ">=1.21.1"
+werkzeug = ">=0.12.2"
+gunicorn = "*"
+
```
Comment (osteele) [inline Pipfile]: You can probably pare this import list down to something like:

```
arrow = ">=0.10.0"
celery = ">=3.1.25"
celery-with-mongodb = ">=3.0"
flask = ">=0.12.2"
flask-cors = ">=3.0.2"
flask-restful = ">=0.3.6"
flask-socketio = ">=2.9.0"
flask-sslify = ">=0.1.5"
icalendar = ">=3.11.4"
isodate = ">=0.5.4"
mccabe = ">=0.6.1"
mongoengine = ">=0.13.0"
pymongo = ">=3.4.0"
python-dateutil = ">=2.6.0"
pytz = ">=2017.2"
requests = ">=2.18.1"
```

(Via `imports=$(rg '^\s*import\s+(\w+)|^\s*from\s+(\w+)\s+import' --no-filename -or '$1$2' | sort | uniq | tr "\n" "|"); egrep ${imports: : -1} Pipfile`.)

File: Pipfile [hunk 2]
```diff
@@ -0,0 +1,53 @@
...
+greenlet = ">=0.4.12"
+honcho = ">=1.0.1"
+icalendar = ">=3.11.4"
+idna = ">=2.5"
+isodate = ">=0.5.4"
+itsdangerous = ">=0.24"
+"jinja2" = ">=2.9.6"
+kombu = ">=3.0.37"
+markupsafe = ">=1.0"
+mccabe = ">=0.6.1"
```
Comment (osteele) [inline Pipfile]: `flake8`, `mccabe`, `pycodestyle`, and `pyflakes` could go under `[dev-packages]`.

File: Pipfile (hunk 1 above)
Comment (osteele) [reply Pipfile]: Moved to #154

File: Pipfile (hunk 2 above)
Comment (osteele) [reply Pipfile]: Moved to #154

**Pull request merged**